ext/pgsql/pgsql.c

```diff
@@ -991,6 +992,12 @@ static void php_pgsql_get_link_info(INTERNAL_FUNCTION_PARAMETERS, int entry_type
 			PQclear(res);
 			return;
 		}
+#if defined(HAVE_PG_SERVICE)
+		case PHP_PG_SERVICE: {
+			result = PQservice(pgsql);
```

Comment:
Can this return NULL? If it does, then it will return the interned empty string, in which case the `@refcount 1` annotation is wrong.

Reply:
Unless there's an exception for interned strings, but I don't believe so? I could be wrong

Reply:
Yes it can return NULL.